code_to_review2.py

```diff
@@ -0,0 +1,106 @@
+import uuid
+from rest_framework import status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+
+class Homeowner:
+    name: str = None
+    email: str = None
+    created_at: str = None
+
+
+class Contractor:
+    name: str = None
+    email: str = None
+    created_at: str = None
+    is_active = None
+
+
+class Project:
+    id: str
+    homeowner: Homeowner
+    contractor: Contractor
+    name: str = None
```

Comment:
Though, this will not result into an error, however, since we are declaring "name" as a string, I would recommend giving it a default value of a string, rather than a None object, just to make things more explicit and easy to understand. The other option is to not assign a default value, at all.

```
name: str = ""
```

Same goes for minBudget and maxBudget

```
minBudget: float = 0.00
maxBudget: float = 0.00
```

Same for the attributes in Homeowner and and Contractor class